pkg/controller/validation/check_snapshot.go

```diff
@@ -16,32 +16,48 @@ import (
 var log = logf.Log.WithName("check_snapshot")
 
 const (
-	snapshotCRDVersion = "v1alpha1"
-	snapshotCRDName    = "volumesnapshots.snapshot.storage.k8s.io"
+	alphaVersion           = "v1alpha1"
+	snapshotCRDName        = "volumesnapshots.snapshot.storage.k8s.io"
+	snapshotClassCRDName   = "volumesnapshotclasses.snapshot.storage.k8s.io"
+	snapshotContentCRDName = "volumesnapshotcontents.snapshot.storage.k8s.io"
 )
 
-// Returns an error if the v1alpha1 VolumeSnapshot CRD is installed in the cluster
+// Returns an error if the v1alpha1 CRD is installed in the cluster
 // and nil otherwise. This is used to prevent upgrading where clusters have manually
 // installed versions we don't support
+
 func CheckAlphaSnapshot(client client.Client) error {
-	snapshotCRD := &apiextv1beta1.CustomResourceDefinition{}
-	err := client.Get(context.TODO(), types.NamespacedName{Name: snapshotCRDName, Namespace: corev1.NamespaceAll}, snapshotCRD)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			// If we can't find the VolumeSnapshot CRD, then we're safe to proceed
-			return nil
+	crdMap := map[string]string{"VolumeSnapshot": snapshotCRDName,
+		"VolumeSnapshotClass":   snapshotClassCRDName,
+		"VolumeSnapshotContent": snapshotContentCRDName}
+	msgString := ""
+	resource := ""
+	found := false
+	crd := &apiextv1beta1.CustomResourceDefinition{}
+	for k, v := range crdMap {
+		err := client.Get(context.TODO(), types.NamespacedName{Name: v, Namespace: corev1.NamespaceAll}, crd)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				errMsg := "Error attempting to obtain existing " + k + " CRD."
+				log.Error(err, errMsg)
+				return err
+			}
```

Comment:
Should you stop processing the CRD on `IsNotFound` error? (i.e. `continue` here, with a comment?)

Reply:
I updated this to log a debug message and then `continue` to skip the rest of the loop.